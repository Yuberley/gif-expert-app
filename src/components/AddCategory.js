import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories, categories, setLimit }) => {

	const [inputValue, setInputValue] = useState('');
	const [limitInput, setLimitInput] = useState('');

	const handleAddInputChange = ( e ) => {
		setInputValue( e.target.value );
	};

	const handleAddILimitChange = ( e ) => {
		setLimitInput( e.target.value );
	};

	const handleSubmit = ( e ) => {
		e.preventDefault();

		if( categories.some( category => category === inputValue ) ){
			const newCategories = categories.filter( category => category !== inputValue);
			setCategories( newCategories );
		}

		if ( inputValue.trim().length > 0 ) {
			setCategories( (cats) => [inputValue, ...cats]);
			setLimit( limitInput );
			setInputValue('');
		}
	};

	return (
		<form onSubmit={ handleSubmit }>
			<div className="container-fluid">
				<div className="row">
					<div className="col-lg-8 col-sm-12">			
						<input
							type="text"
							placeholder=" Search"
							value={ inputValue }
							onChange={ handleAddInputChange }
						/>
					</div>
					<div className="col-lg-2 col-sm-12"></div>
					<div className="col-lg-1 col-sm-6">Quantity</div>
					<div className="col-lg-1 col-sm-6">
						<select name="select" className="form-select" aria-label="Default select example" onChange={ handleAddILimitChange }>
							<option value="10" default>10</option>
							<option value="20">20</option>
							<option value="50">50</option>
							<option value="100">100</option>
						</select>
					</div>
				</div>
			</div>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
