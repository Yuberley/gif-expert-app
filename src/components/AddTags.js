import React from 'react';

export const AddTags = ({ category }) => {
	return (
		<>
			<span className="badge rounded-pill bg-success text-capitalize">{category}</span>
            <span> </span>   
		</>
	);
};
