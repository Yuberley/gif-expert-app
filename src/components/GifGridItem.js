import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
	return (
		<>
			<div className="card animate__animated animate__fadeInUp">
				<img src={ url } alt={ title } />
				<div className="title">
					<p className="title-card" > { title } </p>
				</div>
			</div>
		</>
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
}