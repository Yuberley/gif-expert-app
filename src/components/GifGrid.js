import React from 'react'
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category, limit }) => {

    const { data:gifs, loading } = useFetchGifs( category, limit );

    return (
        <>
            <h3 className="text-capitalize">{ category } {gifs.length === 0 && <small className="text-muted">Not found</small>}</h3>

            { 
                loading &&  <div className="d-flex align-items-center">
                                <strong>Loading...</strong>
                                <div className="spinner-border ms-auto" role="status" aria-hidden="true"></div>
                            </div> 
            }


            <div className="card-grid">
                
                {
                    gifs.map( gif => (
                        <GifGridItem
                            key={ gif.id } 
                            { ...gif } 
                        />
                    ))
                }
                
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limit: PropTypes.string.isRequired,
}