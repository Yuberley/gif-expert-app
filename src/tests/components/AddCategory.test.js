import React from 'react'
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"


describe('Test on component <AddCategory />', () => {

    const setCategories = () => {}
    const setLimit = () => {}
    const categories = [];
    
    const wrapper = shallow( <AddCategory 
                                setCategories={ setCategories } 
                                categories={ categories }
                                setLimit={ setLimit }
                            /> );

    test('should show the component successfully', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('shoul change the input text', () => {

        const input = wrapper.find('input');
        const value = 'Hello world'

        input.simulate('change', {target: {value}})

    })

})