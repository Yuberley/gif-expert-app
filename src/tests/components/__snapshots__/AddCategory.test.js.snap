// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <AddCategory /> should show the component successfully 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-8 col-sm-12"
      >
        <input
          onChange={[Function]}
          placeholder=" Search"
          type="text"
          value=""
        />
      </div>
      <div
        className="col-lg-2 col-sm-12"
      />
      <div
        className="col-lg-1 col-sm-6"
      >
        Quantity
      </div>
      <div
        className="col-lg-1 col-sm-6"
      >
        <select
          aria-label="Default select example"
          className="form-select form-select-sm"
          name="select"
          onChange={[Function]}
        >
          <option
            default={true}
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
          <option
            value="50"
          >
            50
          </option>
          <option
            value="100"
          >
            100
          </option>
        </select>
      </div>
    </div>
  </div>
</form>
`;
